mastodon/src/main/java/org/joinmastodon/android/fragments/ComposeFragment.java

```diff
@@ -1033,16 +1034,23 @@ private void onVisibilityClick(View v){
 		UiUtils.enablePopupMenuIcons(getActivity(), menu);
 		m.setGroupCheckable(0, true, true);
 		m.findItem(switch(statusVisibility){
-			case PUBLIC, UNLISTED -> R.id.vis_public;
+			case PUBLIC -> R.id.vis_public;
+			case UNLISTED -> R.id.vis_unlisted;
 			case PRIVATE -> R.id.vis_followers;
 			case DIRECT -> R.id.vis_private;
 		}).setChecked(true);
+
+		if (statusVisibility != StatusPrivacy.UNLISTED) {
+			m.findItem(R.id.vis_unlisted).setVisible(false);
+		}
 		menu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener(){
 			@Override
 			public boolean onMenuItemClick(MenuItem item){
 				int id=item.getItemId();
-				if(id==R.id.vis_public){
+				if(id==R.id.vis_public) {
```

Comment:
The rest of the code is formatted this way, I recommend opening another issue for deciding on formatting for contributions. (See below lines 1052, 1067 as well)